dask/dataframe/io/parquet/core.py

```diff
@@ -408,6 +409,11 @@ def to_parquet(
         If False (default), construct data-set from scratch. If True, add new
         row-group(s) to an existing data-set. In the latter case, the data-set
         must exist, and the schema must match the input data.
+    overwrite : bool, optional
+        If False (default), write dataset to specified folder at path.  If True, delete
+        all files in the partitioned folder and write a fresh set of partitioned files.
+        Overwrite expects that path already exists, is a non-empty directory, and does
+        not reside in the location from which the function is being called.
```

Comment:
```suggestion
        Whether or not to remove the contents of `path` before writing the dataset.
        The default is False.  If True, the specified path must correspond to
        a directory (but not the current working directory).  This option cannot
        be set to True if `append=True`.
        NOTE: `overwrite=True` will remove the original data even if the current
        write operation fails.  Use at your own risk.
```
I don't feel strongly about the specific wording, but I'd like the functionality to be closer to this discription.  I think it makes sense to allow the user to set `overwrite=True` even if the target directory is empty.  However, we should definitely throw an error if the user tries to clear the current working directory (I'm glad you considered that possibility) or tries to use this option with `append=True`.